### DISCUSSION TITLE
How to resize gif image which has offset and different size for every frame?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#6894] [Development]
### ImageMagick version

7.1.0-52

### Operating system, version and so on

Debian 6.3.0-18+deb9u1

### Description

I have a gif image need to resize, and every frame has different size and offset.
![image](https://github.com/ImageMagick/ImageMagick/assets/10097167/f6086fd9-39a7-4cda-bb51-4b72d4a92339)

I try to resize it like this, and it works, but it takes a lots of time in **CoalesceImages** and **DeconstructImages**, I am not sure if it is the best way to handle resizing of this kind of gif image. Is there any best practice or wiki? Thanks a lot.

```
// Use Golang SDK of IM, mw's type is "*imagick.MagickWand"
mw = mw.CoalesceImages()

mw.ResetIterator()
defer mw.ResetIterator()
	
frames := int(mw.GetNumberImages())

for idx := 0; idx < frames; idx++ {
	mw.SetIteratorIndex(idx)
	w, h, x, y, err := mw.GetImagePage()
	err = mw.ResetImagePage(fmt.Sprintf("%dx%d+%d+%d", w, h, x, y))
	err = mw.ResizeImage(dstW, dstH, param.Filter)
}

mw = mw.DeconstructImages()

// output it
...
```

**Comment by snibgo** [comment]
I suppose your GIF has different offsets and sizes, but the same canvas sizes 720x1280, because the frames are cropped to only the differences between frames. "Coalescing" is the correct method to reverse this process, so every frame becomes the same size, ready for the resize.

I don't use Golang, and can't advise on that.

**Reply by ghost** [reply]
@snibgo Thanks a lot. But CoalesceImages and DeconstructImages will cost a lot of time, is this the only way to handle this kind of gif? is there any faster way? If you use other language SDK, is also ok for me, can you give me some example of right way to resize it? Is there any method can do the resize with this kind of gif without doing the reverse process which will cost a lot of time
This is the example gif. thanks again.
![5](https://github.com/ImageMagick/ImageMagick/assets/10097167/1d5e1bfb-cbfb-45a2-b852-f54e656a247b)
